src/libxrpl/protocol/STTx.cpp

```diff
@@ -187,6 +189,23 @@ STTx::getSeqProxy() const
     if (seq != 0)
         return SeqProxy::sequence(seq);
 
+    if (isFieldPresent(sfBatchTxn))
+    {
+        STObject const batchTxn = const_cast<ripple::STTx&>(*this)
+                                      .getField(sfBatchTxn)
+                                      .downcast<STObject>();
+
+        std::uint32_t const batchIndex{batchTxn.getFieldU8(sfBatchIndex)};
+        if (batchTxn.isFieldPresent(sfTicketSequence))
+        {
+            std::uint32_t const ticketSeq{
+                batchTxn.getFieldU32(sfTicketSequence)};
+            return SeqProxy{SeqProxy::ticket, ticketSeq + batchIndex};
```

Comment:
Shouldn't it just be the ticket sequence? Why are we adding `batchIndex` here?

Reply:
I agree, it should just be the ticket sequence. Tickets may not be chronological.